lib/src/settings.rs

```diff
@@ -155,13 +155,11 @@ impl UserSettings {
 pub struct JJRng(Mutex<ChaCha20Rng>);
 impl JJRng {
     pub fn new_change_id(&self) -> ChangeId {
-        let random_bytes = self.gen();
-        // The `uuid` crate is used merely to specify the number of random bytes (16)
-        ChangeId::from_bytes(
-            uuid::Builder::from_random_bytes(random_bytes)
-                .into_uuid()
-                .as_bytes(),
-        )
+        let mut random_bytes: [u8; 16] = self.gen();
+        // TODO: make it fully random
+        random_bytes[6] = (random_bytes[6] & 0x0f) | ((4 as u8) << 4); // Version: 4
+        random_bytes[8] = (random_bytes[8] & 0x3f) | 0x80; // Variant::RFC4122
+        ChangeId::new(random_bytes.into())
```

Comment:
Can we remove the dependency on the `uuid` crate now then?

Reply:
Sadly there's one more dep: `create_no_gc_ref()`. Does hyphenated syntax matter there?

Reply:
It doesn't matter for any technical reason and I don't think users care either. Feel free to change it as far as I'm concerned. But also feel free to merge this as is.

Reply:
I'll send another PR to drop uuid dependency.